test/helpers/framework.go

```diff
@@ -120,12 +121,89 @@ func (tc *E2ETestFramework) CreateTestNamespace() string {
 		},
 	}
 	_, err := tc.KubeClient.CoreV1().Namespaces().Create(namespace)
-	if err != nil {
+	if err != nil && !errors.IsAlreadyExists(err) {
 		logger.Error(err)
 	}
 	return name
 }
 
+// forceCreate does client.Create(config), deleting an existing instance if necessary.
+// It adds the new object to the cleanup list with deleteOpts
+// Panics if the client or config values are invalid or mismatched.
+func (tc *E2ETestFramework) forceCreate(client, config interface{}, deleteOpts *metav1.DeleteOptions) (newConfig interface{}, err error) {
+	rclient := reflect.ValueOf(client)
+	createMethod := rclient.MethodByName("Create")
+	deleteMethod := rclient.MethodByName("Delete")
+	if !createMethod.IsValid() || !deleteMethod.IsValid() {
+		panic(fmt.Errorf("not a kube client interface: %#v", client))
+	}
+	name := reflect.ValueOf(config).Elem().FieldByName("Name")
+	if !name.IsValid() {
+		panic(fmt.Errorf("not a valid config object, no Name: %#v", config))
+	}
+
+	create := func() (reflect.Value, error) {
+		result := createMethod.Call([]reflect.Value{reflect.ValueOf(config)})
+		return result[0], valueToError(result[1])
+	}
+	delete := func() error {
+		result := deleteMethod.Call([]reflect.Value{name, reflect.ValueOf(deleteOpts)})
+		return valueToError(result[0])
+	}
+	v, err := create()
+	if errors.IsAlreadyExists(err) {
+		if err = delete(); err == nil {
```

Comment:
Don't we need to wait for the delete() to take effect before we attempt the create()?

Reply:
I was assuming that *Interface.Delete() is synchronous, if not then we do need to add something here. Needs to be checked. Another option is to use Update() instead of Delete()+Create() -  I'm unclear if that's equivalent - also needs to be checked.